### DISCUSSION TITLE
counting records slows down the whole application

### DISCUSSION THREAD

**Discussion by Batnik** [laravel/nova-issues#3383] [Q&A] [answered]
select count(*) as aggregate from `my_very_big_table`

it slows down our application a lot, cpu is > 100% all the time, we have many tables with 1-3M records...

what can we do?

**Comment by crynobone** [comment, marked as answer]
I would suggest looking at indexing, Kai Sassnowski made a good talk at Laracon covering this subject.

https://www.youtube.com/watch?v=HubezKbFL7E

**Reply by piterssson** [reply]
The same problem. Database has indexing. Only query "count as aggregate" is very slowly...

![Screenshot from 2021-05-10 10-21-45](https://user-images.githubusercontent.com/16086241/117643190-69956880-b188-11eb-9f5a-ac2ba8abdd69.png)

**Reply by crynobone** [reply]
Seem like query subselect would be a better choice instead of join. https://laravel-news.com/eloquent-subquery-enhancements
